.github/workflows/python.yml

```diff
@@ -95,30 +95,39 @@ jobs:
       - name: Test
         working-directory: py/tests
         run: python3 -m pytest
+      - name: Display structure of dist directory
+        run: ls -R dist
       - name: Upload wheels
-        if: "startsWith(github.ref, 'refs/tags/')"
+        # if: "startsWith(github.ref, 'refs/tags/')"
         uses: actions/upload-artifact@v4
         with:
-          name: wheels
+          # Ensure we have a different name for every artifact, by using the
+          # matrix components in the artifact name
+          name: binary-${{ matrix.type }}-${{ matrix.os }}-${{ matrix.python-version }}
```

Comment:
I think we should at least include something that is different for tag and branch jobs.

Reply:
See above - as I understand it, tag and branch jobs would be in different workflows, and would not have access to each others' artifacts without specifically specifying a `run-id` and adding a `GITHUB_TOKEN` to the action. Does that make sense? Am I missing something?